### PR TITLE
Add otel-collector project

### PR DESCRIPTION
**Pull request by ivyostosh**
*Issue #, if available:* N/A

*Description of changes:* Add otel-collector project

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ivyostosh** [comment]
/hold
This version is not finalized

**Comment by TerryHowe** [comment]
Generally looks good, but obviously that presubmit failed.

**Review by jaxesn — COMMENTED** [review]
Good looks good to me, great work!  Make sure the presubmit is setup and passing and any cdk changes that need to be done have been rolled through prod, before merging

File: projects/aws-observability/aws-otel-collector/Makefile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+IMAGE_COMPONENT=aws-otel-collector/aws-otel-collector
+IMAGE_NAMES=aws-otel-collector
+
+HAS_HELM_CHART=true
+HELM_GIT_TAG:=$(shell cat HELM_GIT_TAG)
+HELM_SOURCE_OWNER=open-telemetry
+HELM_SOURCE_REPOSITORY=opentelemetry-helm-charts
+HELM_DIRECTORY=charts/opentelemetry-collector
+HELM_DESTINATION_REPOSITORY=aws-otel-collector/aws-otel-collector-helm
```
Comment (TerryHowe) [inline projects/aws-observability/aws-otel-collector/Makefile:19]: if we change direction where we are using the controller, it seems like this name would be confusing. maybe no collector in the destination or adot-helm or something? maybe there is another helm chart for that?

**Comment by ivyostosh** [comment]
The team has decided to not build the ADOT image from source, but rather copying the ADOT image to our ECR repo and use that directly. See https://github.com/aws/eks-anywhere-build-tooling/pull/1222 for the new change.

**Pull request closed without merging**